### DISCUSSION TITLE
Response times in the seconds range after moving to a *stronger* server

### DISCUSSION THREAD

**Discussion by Fivefold** [TriliumNext/Trilium#3790] [Q&A] [answered]
First off: sorry if something like this was already asked, I searched but couldn't finde something that fit.

I originally had trilium on a very small vps (1 vcore Intel Xeon E5-2650L @ 1.1 GHz, 2 GB RAM) with very good performance; it was fast and responsive. Because I needed more performance for other applications I moved to a bigger vps (4 vcores Xeon Cascadelake Processor @ 2.1 GHz, 8 GB RAM). I benchmarked both and the latter has - as expected - much better performance in all respects.

The problem I have is that trilium is quite slow here. Actions like opening a subtree or switching notes takes somewhere between 0.5 and 3 seconds, most of the time over one second though. Apps beside trilium are fine. I'm not even sure if this is a _trilium_ problem, an _nginx_ problem or a _server_ problem but hope someone can help me troubleshoot this.

Some notes on my setup (which is the same on both vps', only the trilium version on the old vps is older):
trilium is running in a docker container. Version is the newest (v0.59.3). It sits behind an nginx reverse proxy.

RAM usage is around 2 GB, CPU usage does not rise 15 % average, load average is between 0.6 and 0.8.

Nginx config (I'm using [Nginx proxy manager](https://github.com/NginxProxyManager/nginx-proxy-manager)):
[general](https://user-images.githubusercontent.com/38764218/228810980-ae47f614-7bb5-43e0-9d77-49661449abb9.png)
[SSL](https://user-images.githubusercontent.com/38764218/228811576-4263bf43-d8c5-40a4-9d3f-b19b378bf996.png)

According to the docker logs the requests are actually handled quickly:
```log
200 POST /api/tree/load with 5605 bytes took 8ms
200 POST /api/tree/load with 3543 bytes took 2ms
200 POST /api/tree/load with 3823 bytes took 3ms
200 GET /api/keyboard-shortcuts-for-notes with 2 bytes took 1ms
200 POST /api/tree/load with 7504 bytes took 2ms
200 GET /api/notes/mqn7kackvUK3 with 436 bytes took 3ms
304 GET /api/note-map/mqn7kackvUK3/backlink-count with 11 bytes took 1ms
200 POST /api/tree/load with 21557 bytes took 4ms
200 GET /api/script/startup with 831 bytes took 3ms
200 GET /api/notes/W8k8J4eGzSSl with 579 bytes took 2ms
200 GET /api/notes/BOedF6iHYzq3 with 1445 bytes took 2ms
304 GET /api/note-map/BOedF6iHYzq3/backlink-count with 11 bytes took 1ms
204 PUT /api/options with 0 bytes took 4ms
200 GET /api/notes/Ee1tysShsiPY with 818 bytes took 2ms
204 PUT /api/options with 0 bytes took 2ms
200 GET /api/note-map/Ee1tysShsiPY/backlink-count with 11 bytes took 1ms
200 POST /api/tree/load with 3841 bytes took 4ms
200 GET /api/notes/GGcxACncbRvY with 1448 bytes took 2ms
200 GET /api/note-map/GGcxACncbRvY/backlink-count with 11 bytes took 1ms
204 PUT /api/options with 0 bytes took 3ms
200 PUT /api/special-notes/api-script-launcher with 318 bytes took 6ms
```
This is what it looks like in my browser when I open a few subtrees and switch between notes:
![grafik](https://user-images.githubusercontent.com/38764218/228813091-0c26cad4-5010-4e76-892b-3dabc4dc7513.png)

**Comment by zadam** [comment]
FWIW I haven't seen anything like that yet, but it's probably something outside of trilium. As visible in the trilium logs - requests are processed by trilium itself quickly.

**Reply by Fivefold** [reply]
Thank you. I tried setting up a completely fresh instance and connected directly via http://ip:port/ with the same issue, so at least nginx can be ruled out. Any ideas what else I could try?

Here's the docker-compose file I used:

```yaml
version: '2.1'
services:
  trilium:
    image: zadam/trilium:latest
    container_name: trilium-test
    restart: unless-stopped
    environment:
      - TRILIUM_DATA_DIR=/home/node/trilium-data
      - USER_UID=1000
      - USER_GID=1000
    ports:
      - "12001:8080"
    volumes:
      - /home/trilium:/home/node/trilium-data

```

**Comment by holoword** [comment]
Is the network same as the small vps? Maybe is not the machine speed but the network latency problem? how much ping you connect to the server?

**Reply by Fivefold** [reply]
Thank you for the suggestions!
> Is the network same as the small vps?

Unsure what you mean with network.
If you mean the docker network and/or nginx config: yes, exactly same setup
If you mean the domain name: also the same, I switched over the DNS entries to the new server

I set up a fresh trilium docker instance and connected directly via http://ip:port/ with the same results.

> how much ping you connect to the server?

Results of 10 pings: Minimum 24ms, Maximum 32ms, Average: 26 ms

Same results if I ping the server via domain name, the actual address of my trilium instance or just the IP. All other services on this server are fast and responsive.

**Comment by Fivefold** [comment, marked as answer]
I'm not sure if this actually helped or if the server just "got better", since I've mainly used the desktop app and synced to the server in the meanwhile. But since I had update problems with nginx proxy manager I've since switched over to [traefik](https://github.com/traefik/traefik). I haven't changed anything on the trilium or docker side, so it's either unrelated server changes ~~or the switch from nginx to traefik.~~

My response times in trilium are now in a usable range (sub 1 second) so I will close this for now.

Edit: I forgot I tried accessing directly without the reverse proxy before. So it's unlikely the change from nginx to traefik would have helped...
